openucx/xpmem
# TEST/GTEST: Add sanity tests

## What
Adds sanity userspace tests, starting from porting existing C tests.

## Why ?
We want to enable automated sanities.

## How ?
This includes 3 steps:
- move existing tests to `test/share` directory
- add gtest infra
- add/port tests

Tested:
- compile with and without C++14
- ran `make check`
- ran `cd test/share && ./run.sh`
- ran `./test/gtest/gtest`.
